### DISCUSSION TITLE
Log spam in the Next.js server console

### DISCUSSION THREAD

**Discussion by LufyCZ** [WalletConnect/walletconnect-monorepo#5160] [Developer Support]
Hey, we've been dealing with a lot of errors in the console, coming from WalletConnect's logger, making other development logs hard to read. I don't know what the cause of these errors is, but we haven't encountered any actual problems yet.

![image](https://github.com/WalletConnect/walletconnect-utils/assets/20911347/dcf4bf24-c176-4381-bd0f-b712bb2dc3d1)

I've looked for a way to disable the logger, though it looks like the required options are not passable from `wagmi`.

Wagmi uses the `EthereumProvider`
![image](https://github.com/WalletConnect/walletconnect-utils/assets/20911347/784b12c7-88fc-4f29-a4fc-3bca159cf877)

[which, in turn, uses the](https://github.com/WalletConnect/walletconnect-monorepo/blob/5899f274aaf97f0ae0726d227474655edb526f2e/providers/ethereum-provider/src/EthereumProvider.ts#L560) `UniversalProvider`
![image](https://github.com/WalletConnect/walletconnect-utils/assets/20911347/e137f918-c845-4b76-a195-0fea023c498d)

[where the logger gets set up](https://github.com/WalletConnect/walletconnect-monorepo/blob/5899f274aaf97f0ae0726d227474655edb526f2e/providers/universal-provider/src/UniversalProvider.ts#L65-L68)
![image](https://github.com/WalletConnect/walletconnect-utils/assets/20911347/9917f63d-99bf-4894-8df7-93088782d771)

Unfortunately, there's no way to get the `logger` option set when `UniversalProvider` is initialized `EthereumProvider`, so there's no way to set the logger to `silent`.

This is half bug half feature request, hopefully it's fine like this.

**Comment by linear** [comment]
<p><a href="https://linear.app/walletconnect/issue/CR-95/log-spam-in-the-nextjs-server-console">CR-95 Log spam in the Next.js server console</a></p>

**Comment by glitch-txs** [comment]
> Hey, we've been dealing with a lot of errors in the console, coming from WalletConnect's logger

Hey could you share a way to repro this? Which errors?

**Reply by LufyCZ** [reply]
The first error in the first screenshot, I have traced it back to the pino logger from the WalletConnect logger package. The error usually only prints locally, not when the app is deployed to vercel, not sure why that'd happen.

I can get a minimal repro up in a week or two, don't have time atm.
